tests/cpptests/test_cpp_cursors.cpp

```diff
@@ -111,4 +120,48 @@ TEST_F(CursorsTest, OwnershipAPI) {
   // When cursor is explicitly freed, it should be deleted
   ASSERT_EQ(Cursor_Free(cur), REDISMODULE_OK) << "Cursor should be deleted";
   ASSERT_EQ(Cursors_GetInfoStats().total_user, 0) << "Cursor should be deleted";
+
+  // Case 5: CursorList_Empty on multiple cursors, some idle, some active
+  // Verify that the idle cursors are freed immediately, and the active ones are marked for deletion
+  constexpr int numCursors = 5;
+  constexpr int numIdle = numCursors / 2 + numCursors % 2;
+  uint64_t ids[numIdle] = {0};
+
+  for (int i = 0; i < numCursors; ++i) {
+    cur = Cursors_Reserve(&g_CursorsList, dummy, 1000, NULL);
+    ASSERT_TRUE(cur != NULL);
+    ASSERT_FALSE(cur->delete_mark);
+    ASSERT_FALSE(is_Idle(cur));
+    if (i % 2 == 0) {
+      ASSERT_EQ(Cursor_Pause(cur), REDISMODULE_OK) << "Cursor should be paused";
+      ids[i % 2] = cur->id;
```

Comment:
currently only setting index `0`
```suggestion
      ids[i / 2] = cur->id;
```
Also, consider using a `std::vector` or a `std::set`, then just push back